<?xml version="1.0"?>
<launch>
  <arg name="no_static_map" default="false"/>

  <arg name="base_global_planner" default="global_planner/GlobalPlanner"/>
  <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
  <!-- <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/> -->

  <node pkg="move_base" type="move_base" respawn="true" name="move_base" output="screen">

    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    <rosparam file="$(find main_package)/config/planner.yaml" command="load"/>

    <!-- observation sources located in costmap_common.yaml -->
    <rosparam file="$(find main_package)/config/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find main_package)/config/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- local costmap, needs size -->
    <rosparam file="$(find main_package)/config/costmap_local.yaml" command="load" ns="local_costmap" />
    <param name="local_costmap/width" value="5.0"/>
    <param name="local_costmap/height" value="5.0"/>
    <param name="local_costmap/resolution" value="0.05"/>

    <!-- static global costmap, static map provides size -->
    <!-- <rosparam file="$(find main_package)/config/costmap_global_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/> -->
    <rosparam file="$(find main_package)/config/costmap_slope_static.yaml" command="load" ns="global_costmap" unless="$(arg no_static_map)"/>
    
    <!-- global costmap with laser, for odom_navigation_demo -->
    <!-- <rosparam file="$(find main_package)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>    -->
    <param name="global_costmap/width" value="500.0" />
    <param name="global_costmap/height" value="500.0" />
    <param name="global_costmap/resolution" value="0.1"/>
    
    <param name="max_planning_retries" value="10"/>
    
    <remap from="odom" to="/odometry/filtered_fusion"/>
    <remap from="cmd_vel" to="/r2d2_diff_drive_controller/cmd_vel"/>
  </node>

</launch>
